src/main/java/com/flowpowered/noise/module/Module.java

```diff
@@ -26,44 +26,115 @@
  */
 package com.flowpowered.noise.module;
 
-import com.flowpowered.noise.exception.NoModuleException;
+import com.flowpowered.noise.module.combiner.*;
+import com.flowpowered.noise.module.misc.Cache;
+import com.flowpowered.noise.module.modifier.*;
+import com.flowpowered.noise.module.combiner.selector.*;
 
+import java.util.Collection;
+
+/**
+ * Represents a function from {@code R^3 -> R}, or taking three doubles and returning one. The
+ * return value should be in the range {@code [0, 1]}. In this way, one can consider a Module
```

Comment:
The range is only true for the functions in the Noise class. The generators can have a larger range because of octaves. You should remove this.